app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt

```diff
@@ -1,7 +1,33 @@
 package com.example.weatherandroidapp.viewmodels
 
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.domain.entities.ResponseWeatherInfo
+import com.example.domain.usecase.UseCaseInterfaceGetCity
+import com.example.domain.utils.ServiceResult
+import com.example.weatherandroidapp.contracts.WeatherGridContract
+import com.example.weatherandroidapp.utils.Data
+import com.example.weatherandroidapp.utils.Event
+import com.example.weatherandroidapp.utils.Status
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class WeatherGridViewModel : ViewModel() {
-    // TODO: Implement the ViewModel
+class WeatherGridViewModel(
+    private val useCaseWeatherCity: UseCaseInterfaceGetCity
+) : ViewModel(), WeatherGridContract.ViewModel {
+
+    val mutableMainState: MutableLiveData<Event<Data<ResponseWeatherInfo>>> = MutableLiveData()
```

Comment:
please make this private and add a publc get. also give it a more expecific name please

Reply:
Doing

Reply:
also you can use lazy init for this var